staging/src/k8s.io/kms/service/grpc_service.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"k8s.io/klog/v2"
+	kmsapi "k8s.io/kms/apis/v2alpha1"
+)
+
+// GRPCService is a gprc server that runs the kms v2 alpha 1 API.
+type GRPCService struct {
+	addr    string
+	timeout time.Duration
+	server  *grpc.Server
+
+	kmsService Service
+}
+
+var _ kmsapi.KeyManagementServiceServer = (*GRPCService)(nil)
+
+// NewGRPCService creates an instance of GRPCService.
+func NewGRPCService(
+	address string,
+	timeout time.Duration,
+
+	kmsService Service,
+) (*GRPCService, error) {
+	klog.V(4).Infof("Configure KMS plugin with endpoint: %s", address)
+
+	s := &GRPCService{
+		addr:       address,
+		timeout:    timeout,
+		kmsService: kmsService,
+	}
+
+	return s, nil
+}
+
+// ListenAndServe accepts incoming connections on a Unix socket. Returns non-nil
+// error unless Close or Shutdown is called.
+func (s *GRPCService) ListenAndServe() error {
+	ln, err := net.Listen("unix", s.addr)
+	if err != nil {
+		return err
+	}
+	defer ln.Close()
+
+	gs := grpc.NewServer(
+		grpc.ConnectionTimeout(s.timeout),
+	)
+	s.server = gs
+
+	kmsapi.RegisterKeyManagementServiceServer(gs, s)
+
+	return gs.Serve(ln)
+}
+
+// Shutdown performs a grafecul shutdown. Doesn't accept new connections and
+// blocks until all pending  RPCs are finished.
+func (s *GRPCService) Shutdown() {
+	if s.server != nil {
+		s.server.GracefulStop()
```

Comment:
If this returns an error, ignore it explicitly.

Reply:
`s.server.GracefulStop()` has the signature: `func (s *Server) GracefulStop()`.